components/brave_shields/browser/tracking_protection_service.h

```diff
@@ -43,14 +47,51 @@ class TrackingProtectionService : public BaseLocalDataFilesObserver {
                         const base::FilePath& install_dir,
                         const std::string& manifest) override;
 
+#if BUILDFLAG(BRAVE_STP_ENABLED)
+  bool ShouldStoreState(HostContentSettingsMap* map,
+    int render_process_id, int render_frame_id, const GURL& top_origin_url,
+    const GURL& origin_url);
+
+  void SetStartingSiteForRenderFrame(GURL starting_site, int render_process_id,
+    int render_frame_id);
+  GURL GetStartingSiteForRenderFrame(int render_process_id,
+    int render_frame_id);
+  void DeleteRenderFrameKey(int render_process_id, int render_frame_id);
+  void ModifyRenderFrameKey(int old_render_process_id, int old_render_frame_id,
+    int new_render_process_id, int new_render_frame_id);
+#endif
+
+ protected:
+#if BUILDFLAG(BRAVE_STP_ENABLED)
+  void ParseStorageTrackersData();
+  struct RenderFrameIdKey {
+    RenderFrameIdKey();
+    RenderFrameIdKey(int render_process_id, int frame_routing_id);
+
+    int render_process_id;
+    int frame_routing_id;
+
+    bool operator<(const RenderFrameIdKey& other) const;
+    bool operator==(const RenderFrameIdKey& other) const;
+  };
+#endif
+
  private:
   void OnDATFileDataReady();
   std::vector<std::string> GetThirdPartyHosts(const std::string& base_host);
 
+#if BUILDFLAG(BRAVE_STP_ENABLED)
+  std::vector<std::string> first_party_storage_trackers_;
```

Comment:
Why do we make a linear search over a vector? Maybe use `base::flat_map` or `std::map`?
https://chromium.googlesource.com/chromium/src/base/+/master/containers/README.md

Reply:
Really nice constructs, was not aware about this. Looks like `flat_set` is more suitable here, let me know if you think `flat_maps` are better?